game/src/main/kotlin/world/gregs/voidps/world/interact/entity/npc/combat/type/GiantMole.kts

```diff
@@ -0,0 +1,209 @@
+import world.gregs.voidps.engine.entity.World
+import world.gregs.voidps.engine.entity.character.face
+import world.gregs.voidps.engine.entity.character.facing
+import world.gregs.voidps.engine.entity.character.mode.move.enterArea
+import world.gregs.voidps.engine.entity.character.mode.move.exitArea
+import world.gregs.voidps.engine.entity.character.move.tele
+import world.gregs.voidps.engine.entity.character.npc.NPC
+import world.gregs.voidps.engine.entity.character.player.Player
+import world.gregs.voidps.engine.entity.character.player.Players
+import world.gregs.voidps.engine.entity.character.player.skill.Skill
+import world.gregs.voidps.engine.entity.character.setAnimation
+import world.gregs.voidps.engine.entity.obj.objectOperate
+import world.gregs.voidps.engine.entity.playerSpawn
+import world.gregs.voidps.engine.inject
+import world.gregs.voidps.engine.inv.inventory
+import world.gregs.voidps.engine.inv.itemChange
+import world.gregs.voidps.engine.inv.transact.operation.ReplaceItem.replace
+import world.gregs.voidps.engine.map.collision.random
+import world.gregs.voidps.type.Direction
+import world.gregs.voidps.type.Tile
+import world.gregs.voidps.world.interact.entity.combat.*
+import world.gregs.voidps.world.interact.entity.combat.hit.combatAttack
+import world.gregs.voidps.world.interact.entity.gfx.areaGraphic
+import world.gregs.voidps.world.interact.entity.player.equip.inventoryOption
+import world.gregs.voidps.world.interact.entity.sound.areaSound
+import kotlin.random.Random
+
+val logger = InlineLogger()
+val areas: AreaDefinitions by inject()
+val players: Players by inject()
+
+val acceptedTiles = listOf(
+    Tile(3005, 3376, 0),
+    Tile(2999, 3375, 0),
+    Tile(2996, 3377, 0),
+    Tile(2989, 3378, 0)
+)
+
+val giantMoleLair = areas["giant_mole_lair"]
+val gianMoleSpawns = areas["giant_mole_spawn_area"]
+val initialCaveTile: Tile = Tile(1752, 5237, 0)
+
+inventoryOption("Dig") {
+    val playerTile: Tile = player.tile
+    if(!acceptedTiles.contains(playerTile)) {
+        return@inventoryOption
+    }
+    player.setAnimation("dig_with_spade")
+    player.open("warning_dark")
+}
+
+objectOperate("Climb", "giant_mole_lair_escape_rope") {
+    player.setAnimation("climb_up")
+    player.tele(acceptedTiles.random())
+}
+
+interfaceOption(component = "proceed", id = "warning_dark") {
+    player.tele(initialCaveTile, clearInterfaces = true)
+}
+
+interfaceOption(component = "stayout", id = "warning_dark") {
+    player.closeInterfaces()
+}
+
+combatAttack {
+    val npc = target as NPC
+    if(npc.id == "giant_mole") {
+        val currentHealth = npc.levels.get(Skill.Constitution)
+        var shouldBurrow = false
+        if (it.fightStyle == "magic" && damage != 0) {
+            shouldBurrow = shouldBurrowAway(currentHealth)
+        } else if (it.fightStyle != "magic") {
+            shouldBurrow = shouldBurrowAway(currentHealth)
+        }
+        if(shouldBurrow && !World.timers.contains("await_mole_burrowing")) {
```

Comment:
Mixing timers and queue's? `World.queue("await_mole_burrowing", 1) {`

I think what you actually want here is an [npc clock](https://github.com/GregHib/void/wiki/clocks):

`npc.start("await_mole_burrowing", 1)`, `npc.hasClock("await_mole_burrowing")`